File: src/Wallet.Entities/EntityObjects/Person.cs
using Wallet.Entities.Base.Concrete;

namespace Wallet.Entities.EntityObjects;

public class Person : SoftDeleteEntity
{
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public string? MiddleName { get; set; }
    public DateTime DateOfBirth { get; set; }
    public required string Gender { get; set; }
    public Guid? NationalityId { get; set; }
    public string? TaxNumber { get; set; }
    public string? IdNumber { get; set; }
    public string? ProfilePictureUrl { get; set; }
    
    // Preferences
    public string? Language { get; set; }
    public string? TimeZone { get; set; }
    public string? Currency { get; set; }
    
    // Navigation Properties
    public ICollection<PersonAddress>? Addresses { get; set; }
    public ICollection<PersonContact>? Contacts { get; set; }
    public User? User { get; set; }  // One-to-One relationship with User
    public Nationality? Nationality { get; set; }
} 

File: src/wallet.ui/src/types/person.ts
export interface PersonData {
    id: string;
    firstName: string;
    lastName: string;
    middleName?: string;
    dateOfBirth: string;
    gender: string;
    language?: string;
    timeZone?: string;
    currency?: string;
    addresses: PersonAddress[];
    contacts: PersonContact[];
}

export interface PersonAddress {
    id: string;
    addressType?: string;
    addressName?: string;
    addressLine1?: string;
    addressLine2?: string;
    district?: string;
    city?: string;
    state?: string;
    country: string;
    postalCode: string;
    isDefault: boolean;
}

export interface PersonContact {
    id: string;
    contactType?: string;
    contactName?: string;
    contactValue?: string;
    countryCode?: string;
    areaCode?: string;
    isDefault: boolean;
    isPrimary: boolean;
} 